### PR TITLE
feat(vps): add tracking to migration of vps

### PR DESCRIPTION
**Pull request by varun257**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/vps-migrate-page-modal`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-7063
| License          | BSD 3-Clause

Add tracking to vps migration

### CODE REVIEW

**Comment by marie-j** [comment]
Hello @varun257 , 
Could you please solve existing conflicts ?
Thanks

File: packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js [hunk 1]
```diff
@@ -9,8 +9,25 @@ export default /* @ngInject */ ($stateProvider) => {
...
-      goToMigrateConfirm: /* @ngInject */ ($state) => () =>
-        $state.go('vps.detail.dashboard.migrate.confirm'),
+      migrationTrackingPrefix: /* @ngInject */ (vpsMigration) => {
+        return `vps::vps-migration::from_${vpsMigration.currentPlan}_to_${vpsMigration.newPlan}`;
+      },
+      trackPage: /* @ngInject */ (atInternet, migrationTrackingPrefix) =>
+        atInternet.trackPage({
+          name: migrationTrackingPrefix,
+          type: 'navigation',
+        }),
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js:19]: Isn't this already handled by `ng-at-internet-ui-router-plugin` ?

File: packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js [hunk 2]
```diff
@@ -32,5 +49,8 @@ export default /* @ngInject */ ($stateProvider) => {
         return promise;
       },
     },
+    atInternet: {
+      ignore: true,
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js:53]: Maybe we could use the 
```suggestion
      rename: <trackingValue>,
```
instead

File: packages/manager/modules/vps/src/dashboard/migrate/migrate.service.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export default class VpsMigrationService {
+  /* @ngInject */
+  constructor($q, $http) {
+    this.$q = $q;
+    this.$http = $http;
+  }
+
+  migrateVps(serviceName) {
+    return this.$http.post(`/vps/${serviceName}/migration2016`);
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/migrate/migrate.service.js:9]:
```suggestion
    return this.$http.post(`/vps/${serviceName}/migration2016`)
    .then(({data}) => data)
```

File: packages/manager/modules/vps/src/dashboard/migrate/migrate.service.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export default class VpsMigrationService {
+  /* @ngInject */
+  constructor($q, $http) {
+    this.$q = $q;
```
Comment (marie-j) [inline packages/manager/modules/vps/src/dashboard/migrate/migrate.service.js:4]:
```suggestion

```

**Comment by radireddy** [comment]
You have commits that are not relevant tracking. Can you rebase and only include commits that are relevent.

File: packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js (hunk 2 above)
Comment (varun257) [reply packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js:53]: I did try but i cannot inject the values which are `resolved` and i do not want to send the `planCodes` as `stateParams`. So i had to ignore here and add page tracking on `resolve/ngInit` instead.

File: packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js (hunk 1 above)
Comment (varun257) [reply packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js:19]: Need customized page tracking here so i am overriding atInternet, ignoring default tracking.

File: packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js [hunk 5]
```diff
@@ -9,8 +9,25 @@ export default /* @ngInject */ ($stateProvider) => {
       },
     },
     resolve: {
-      goToMigrateConfirm: /* @ngInject */ ($state) => () =>
-        $state.go('vps.detail.dashboard.migrate.confirm'),
+      migrationTrackingPrefix: /* @ngInject */ (vpsMigration) => {
```
Comment (radireddy) [inline packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js:12]: you are using this in multiple routes. Can you put it in parent route and reuse it?

Comment (varun257) [reply packages/manager/modules/vps/src/dashboard/migrate/migrate.routing.js:12]: it is 2 different tracking prefixes which will be provided to single component so i am using it in 2 different places.

**Review by Heiiji — APPROVED** [review]

**Review by radireddy — APPROVED** [review]

**Pull request merged**